src/OrleansRuntime/GrainDirectory/ClientObserverRegistrar.cs

```diff
@@ -102,6 +103,7 @@ internal void ClientDropped(GrainId clientId)
 
         private async Task OnClientRefreshTimer(object data)
         {
+            if (gateway == null) return;
```

Comment:
Yes, you are correct. Still, one can never be too cautious. The other code may change in the future, so this is double protection.